plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractIcebergTableOperations.java

```diff
@@ -213,21 +215,20 @@ protected void refreshFromMetadataLocation(String newLocation)
             return;
         }
 
-        AtomicReference<TableMetadata> newMetadata = new AtomicReference<>();
-        Tasks.foreach(newLocation)
-                .retry(20)
-                .exponentialBackoff(100, 5000, 600000, 4.0)
-                .stopRetryOn(org.apache.iceberg.exceptions.NotFoundException.class) // qualified name, as this is NOT the io.trino.spi.connector.NotFoundException
-                .run(metadataLocation -> newMetadata.set(
-                        TableMetadataParser.read(fileIo, io().newInputFile(metadataLocation))));
+        TableMetadata newMetadata = Failsafe.with(new RetryPolicy<>()
+                        .withMaxRetries(20)
+                        .withBackoff(100, 5000, MILLIS, 4.0)
+                        .withMaxDuration(Duration.ofMillis(600000))
```

Comment:
Can we change to `Duration.ofMinutes` so that we can understand the duration quickly?